Common/Statistics/AssetsUnderManagementCapacityManager.cs

```diff
@@ -0,0 +1,238 @@
+        /// The whole method body is locked, since order events can be triggered at any time in live mode
+        /// </remarks>
+        private void Update(object sender, IBaseData data, int orderId)
+        {
+            lock (_lock)
+            {
+                try
+                {
+                    var symbol = data.Symbol;
+                    var security = _portfolio.Securities[symbol];
+
+                    var holdingsTurnover = 1m;
+                    var holdingsValue = security.Holdings.AbsoluteHoldingsValue;
+
+                    if (holdingsValue != 0)
+                    {
+                        var order = _portfolio.Transactions.GetOrderById(orderId);
+                        var orderValue = Math.Abs(order.GetValue(security));
+                        holdingsTurnover = orderValue / holdingsValue;
+
+                        if (holdingsTurnover == 0)
+                        {
+                            RemoveConsolidator(sender, symbol);
+                            return;
+                        }
+                    }
+
+                    var totalMarketVolume = GetTotalMarketVolume(data);
+                    var totalTradeVolumeCapacity = _maximumTradeableVolume * totalMarketVolume / holdingsTurnover;
+
+                    // If it is a new day, we discard the previous day data
+                    var utcDate = data.EndTime.ConvertToUtc(security.Exchange.TimeZone).Date;
+                    if (_previousInputTime != utcDate)
+                    {
+                        _previousInputTime = utcDate;
+                        _maximumSymbolCapacity.Clear();
+                        _historicalPortfolioCapacity.Add(0m);
+                    }
+
+                    // Updates AUM Capacity if there is new information or lower total trade volume capacity for the security
+                    decimal capacity;
+                    if (!_maximumSymbolCapacity.TryGetValue(symbol, out capacity) ||
+                        totalTradeVolumeCapacity < capacity)
+                    {
+                        _maximumSymbolCapacity[symbol] = totalTradeVolumeCapacity;
+                        _historicalPortfolioCapacity[0] = _maximumSymbolCapacity.Sum(x => x.Value);
+                    }
+
+                    RemoveConsolidator(sender, symbol);
+                }
+                catch (Exception e)
+                {
+                    Log.Error($"AssetsUnderManagementCapacityManager.Update: {e}");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the total market volume which is a fraction of the current trade bar volume
+        /// </summary>
+        /// <param name="data">Last price trade bar available or consolidated</param>
+        private static decimal GetTotalMarketVolume(IBaseData data)
+        {
+            var tradeBar = (TradeBar) data;
+
+            var factor = tradeBar.Period == Time.OneDay
+                ? .025m
+                : .050m;
```

Comment:
A comment here describing _why_ we're using `0.025m` and `0.050m` -- in fact, I'm not sure I understand what these factors mean.

Reply:
This factor is the maximum percentage of market volume that is tradeable. These values are ad-hoc defined by research.